src/app/pages/charging-stations/table-actions/table-request-ocpp-params-action.ts

```diff
@@ -0,0 +1,58 @@
+import { Router } from '@angular/router';
+import { TranslateService } from '@ngx-translate/core';
+import { TableAction } from 'app/shared/table/actions/table-action';
+import { ChargingStation, ChargingStationButtonAction, OCPPGeneralResponse } from 'app/types/ChargingStation';
+import { ActionResponse } from 'app/types/DataResult';
+import { ButtonColor, ButtonType, TableActionDef } from 'app/types/Table';
+import { Utils } from 'app/utils/Utils';
+import { Observable } from 'rxjs';
+
+import { CentralServerService } from '../../../services/central-server.service';
+import { DialogService } from '../../../services/dialog.service';
+import { MessageService } from '../../../services/message.service';
+import { SpinnerService } from '../../../services/spinner.service';
+
+export class TableRequestOCPPParamsAction implements TableAction {
+  private action: TableActionDef = {
+    id: ChargingStationButtonAction.REQUEST_OCPP_PARAMS,
+    type: 'button',
+    icon: 'vertical_align_bottom',
+    color: ButtonColor.PRIMARY,
+    name: 'chargers.button_get_configuration',
+    tooltip: 'chargers.button_get_configuration',
+    action: this.requestOCPPParameters,
+  };
+  public getActionDef(): TableActionDef {
+    return this.action;
+  }
+
+  private requestOCPPParameters(chargingStation: ChargingStation, dialogService: DialogService, translateService: TranslateService,
+      messageService: MessageService, centralServerService: CentralServerService, router: Router, spinnerService: SpinnerService,
+      refresh?: () => Observable<void>) {
+    dialogService.createAndShowYesNoDialog(
+      translateService.instant('chargers.get_configuration_title'),
+      translateService.instant('chargers.get_configuration_confirm', {chargeBoxID: chargingStation.id})
+    ).subscribe((result) => {
+      if (result === ButtonType.YES) {
+        spinnerService.show();
+        centralServerService.requestChargingStationOcppParameters(chargingStation.id).subscribe((response: ActionResponse) => {
+          spinnerService.hide();
+          if (response.status === OCPPGeneralResponse.ACCEPTED) {
+            messageService.showSuccessMessage(
+              translateService.instant('chargers.ocpp_params_update_from_template_success', { chargeBoxID: chargingStation.id }));
+            if (refresh) {
+              refresh().subscribe();
+            }
+          } else {
+            Utils.handleError(JSON.stringify(response),
+              messageService, 'chargers.ocpp_params_update_from_template_error');
```

Comment:
The message says: 'Error occurred while updating the OCPP parameters of the charging station' 
But you only request it from the station right?

Reply:
changed messge to reflect actual process